### PR TITLE
feat: add compatibility matrix and OTA info for React Native TS SDK

### PR DESCRIPTION
**Pull request by justin-fiedler**
## Description

https://pr-865.d19s7xzcva2mw3.amplifyapp.com/data/sdks/typescript-react-native/

* add [compatibility matrix](https://pr-865.d19s7xzcva2mw3.amplifyapp.com/data/sdks/typescript-react-native/#compatibility-matrix) and [OTA info](https://pr-865.d19s7xzcva2mw3.amplifyapp.com/data/sdks/typescript-react-native/#over-the-air-updates-ota) for React Native TS SDK

### CODE REVIEW

File: docs/data/sdks/typescript-react-native/index.md [hunk 1]
```diff
@@ -19,6 +19,16 @@ The React Native SDK lets you send events to Amplitude. This library is open-sou
 
 --8<-- "includes/sdk-rn/rn-notification.md"
 
+## Compatibility Matrix
+
+The following matrix lists the support for Amplitude React Native SDK version for different versions of React Native and React Native CLI.
+
+| @amplitude/analytics-react-native | @react-native-community/cli | react-native      |Gradle|Android Gradle Plugin|
+|-----------------------------------|-----------------------------|-------------------|---|---|
+| >= 1.0.0                          | >= 10.0                     | >= 0.71           | 7.5.1+ | 7.2.1+ |
```
Comment (falconandy) [inline docs/data/sdks/typescript-react-native/index.md:28]: I am wondering if the versions should be consistent with:
https://github.com/amplitude/Amplitude-TypeScript/blob/main/packages/analytics-react-native/package.json#L73
https://github.com/amplitude/Amplitude-TypeScript/blob/main/packages/analytics-react-native/android/build.gradle#L9

**Review by falconandy — CHANGES_REQUESTED** [review]

File: docs/data/sdks/typescript-react-native/index.md (hunk 1 above)
Comment (justin-fiedler) [reply docs/data/sdks/typescript-react-native/index.md]: Thanks. Fixed.

**Review by Mercy811 — APPROVED** [review]
Thank you Justin. This is super helpful 🙏

File: docs/data/sdks/typescript-react-native/index.md [hunk 2]
```diff
@@ -743,3 +754,13 @@ The React Native SDK does not directly access the IDFA as it would require addin
 Here is an [example Plugin that sets the IDFA](https://github.com/amplitude/Amplitude-TypeScript/blob/main/examples/plugins/react-native-idfa-plugin/idfaPlugin.ts)  using a third-party library.
 
 --8<-- "includes/abbreviations.md"
+
+### Over the air updates (OTA)
+
+If you are using platform like Expo that supports OTA updates. It is important to know our SDK has both native and JS code. If you are using OTA updates, you will need to make sure the native code is updated as well. See Expo's documentation on [publishing](https://docs.expo.dev/archive/classic-updates/publishing) and [runtime versions](https://docs.expo.dev/eas-update/runtime-versions/) for more details.
```
Comment (yuhao900914) [inline docs/data/sdks/typescript-react-native/index.md:760]: 👍

Comment (yuhao900914) [inline docs/data/sdks/typescript-react-native/index.md:760]: I didn't get the meaning of this actually. Does that mean, we need to use 1.3.0 while using EXPO?

Comment (justin-fiedler) [reply docs/data/sdks/typescript-react-native/index.md:760]: The versions in the list have changes to the native code. Other versions will handle OTA updates without issues.
So if a customer is using `1.1.0` and then want to update to `1.3.0` they can't do it via OTA. But updating to `1.1.0` -> `1.2.0` would be fine. OR from `1.3.0` -> `1.4.0` etc

Comment (yuhao900914) [reply docs/data/sdks/typescript-react-native/index.md:760]: Thanks for the explanation.

**Pull request merged**